com.unity.render-pipelines.high-definition/Editor/RenderPipeline/HighDefinitionRenderPipelineProjectorEditor.cs

```diff
@@ -0,0 +1,22 @@
+using UnityEngine;
+using UnityEngine.Rendering.HighDefinition;
+
```

Comment:
Regarding file name, use the shortcut:

HighDefinitionRenderPipelineProjectorEditor -> HDProjectorEditor

Note: I suppose you put the name of the pipeline in class name to avoid collision when the two pipe are present?
There is namespace here UnityEditor.Rendering.HighDefinition. So question, do we really need to have different name?

Guess typeof(HDRenderPipelineAsset) is enough to not collide, so we could name both editor in both package 
ProjectorEditor.

Last question, do we need two files at all?
Maybe we can put the editor declaration in Core package without specifiy the type of RP asset ?
 [CustomEditorForRenderPipeline(typeof(Projector))]
 And it will apply to all pipeline?
 
 (cc @jenniferd-unity )

Reply:
Note: I am not a UI guy, so it is just random guess

Reply:
Thanks for the comment. I tried to follow existing naming conventions, but I agree the names are a bit convoluted already. I would be fine with just using "ProjectorEditor" for both since they are in different namespaces.

I initially thought about putting it in RP Core, but then you can't customize the error message based on which RP is active (since UniversalRenderPipelineAsset and HDRenderPipelineAsset are not necessarily available). Unless there's some other way to do that neatly.

We could also decide that the RP-specific message is not required, in which case we could have it in RP Core. Originally suggested text included the conditional "... If HDRP is used, use the Decal Projector component instead" which is less clean but conveys the same information.

Reply:
**Maybe we can put the editor declaration in Core package without specifiy the type of RP asset ?**

This is valid concern, it would also make the editor consistent for all SRP, including custom.

Reply:
I agree with the renaming. Plus, up to now, when we named HDxxx or Universalxxx, it was because xxx was something in HDRP or UniversalRP. But here it is not the case. We just override for clarification.

And if we want to deactivate it only for HDRP and UniversalRP, this is ok. But if we want to deactivate it also for any CustomRP, I think Seb as a point here. What is the intent regarding custom SRP?

Reply:
For customization of message, if all in CoreRP, you can grab the render pipeline asset currently used and check its type I guess.

Reply:
Maybe someone has better info about this, but I've understood "Projector" is bit of a legacy feature. So if neither URP nor HDRP support it, it would make sense to me that by default custom SRP don't support it either. I guess they could always choose to write a custom inspector that enables the UI if really needed.

Sounds to me like benefits of having only one custom editor for all SRPs are bigger than the drawback of not being able to tailor the message specifically for HDRP.

Reply:
yup, we are all align :)